extensions/security/runtime-spi/src/main/java/io/quarkus/security/spi/runtime/SecurityEventProducer.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.security.spi.runtime;
+
+/**
+ * Common producer of all security events. This interface can be injected as CDI bean.
+ * All security events should be fired with this producer, so that extensions and
+ * applications have one way to deal with events.
+ */
+public interface SecurityEventProducer {
+
+    /**
+     * Way to notify event observers. Two type of security event observers, if present, will be notified:
+     * - Observer of {@code T} type
+     * - Observer of {@code SecurityEvent} type
+     *
+     * @param securityEvent security event
+     */
+    <T extends SecurityEvent> void fire(T securityEvent);
```

Comment:
There are many points for this API:

1. If you need to disable observing of only security events during runtime, how do you  do it without having one place that takes care of security events?
2. If you prefer publish events with Kafka or anything else, you don't need to first use CDI events and consume them, you can define `SecurityEventProducer` that can filter / transform / publish. I don't expect users do this though.
3. Did you see code I deleted from OIDC extension? Just imagine that his sort of code would need to be repeated anywhere security events are fired, just this PR fires events from: Security, Vert.x HTTP, RESTEasy Reactive, RESTEasy Classic, OIDC

Reply:
but TBH main reason was @sberyozkin comment here https://github.com/quarkusio/quarkus/issues/26549#issuecomment-1828400971, I didn't want to write code just to rewrite it when we _introduce SPI_. This API provides one way to fire security events. `fireAuthenticationEvent` wouldn't be part of the `independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java:53` so would it work the way that extensions just fire event and Arc code only handles it when it is used? Difference is that it means extra work, often on `Uni` I need to call extra `invoke` etc. and create event.

Reply:
> This API provides one way to fire security events. `fireAuthenticationEvent` wouldn't be part of the `independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java:53` so would it work the way that extensions just fire event and Arc code only handles it when it is used?

What I had in mind is something like:
```java
interface SecurityEventProducer {
   void fireAuthenticationEvent(AuthenticationEvent event);
   void fireAuthorizationEvent(AuthorizationEvent event);
}
```
And then the impl bean could inject `Event<AuthenticationEvent>` and `Event<AuthorizationEvent>` and rely on internal optimizations. Ofc, this would not be possible if the set of security events is "unbounded".